rpadmin/admin.go

```diff
@@ -630,3 +632,48 @@ func MaxRetries(r int) Opt {
 		cl.MaxRetries = r
 	}}
 }
+
+// AdminAddressesFromK8SDNS attempts to deduce admin API URLs
+// based on Kubernetes DNS resolution.
+// We are in Kubernetes.
+// https://github.com/kubernetes/dns/blob/master/docs/specification.md
+// Assume that Admin API URL configured is a Kubernetes Service URL.
+// This Admin API URL is passed in as the function argument.
+// Since it's a Kubernetes service, Kubernetes DNS creates a DNS SRV record
+// for the admin port mapping.
+// We can query the DNS record to get the target host names and ports.
+func AdminAddressesFromK8SDNS(adminAPIURL string) ([]string, error) {
```

Comment:
What do you think about making this a method of the `AdminAPI` rather than a standalone function? What I had in mind is that it uses the already configured URLs, and that you have a method such as 

```
func (a *AdminAPI) UpdateAPIUrlsFromKubernetesDNS(ctx context.context) error {}
```

and then it uses https://github.com/redpanda-data/common-go/blob/d034f64565442d1ca943efa0a420952db71a860d/rpadmin/admin.go#L84 as the initial URLs and will also mutate this set of URLs after successful discovery. Would that work and make sense @r-vasquez ?

My assumption is that this method has to be called oftentimes before issuing broker-specific adminapi requests, to ensure we have an up-to-date map and not only once when initializing the adminapi client.

Reply:
Hmm interesting. I guess the workflow I had in mind was that the client (user of the API), would call `AdminAddressesFromK8SDNS` to get candidate URLs, and then call `NewHostClient()` to create a broker-specific Admin API Client. I am open to different approach.

Reply:
I believe if this is becomes a more common usecase (besides the debug bundle work), this can quickly become cumbersome for the user for the admin api client. Therefore, I advocate for putting more logic inside this client, so that it becomes more handy to use.

**One option for that may be:**

Pros:
- Easy to use

Cons:
- Partial responses can become tricky in terms of error handling (see franz-go with sharded requests and error handling)

```go
  // Update internally stored URLs, so that we can talk to all brokers
  adminCl. UpdateAPIUrlsFromKubernetesDNS(ctx)

  // Get debug bundle status for one specific broker
  singleBrokerStatus, err := adminCl.GetDebugBundleStatus(brokerID)

  // Get debug bundle status for multiple brokers. Use nil to get status from all brokers
  statuses, err := adminCl.GetDebugBundleStatuses(brokerIDs)
```

**Another option:**

Pros:
- More generic approach for the adminapi client

Cons:
- You still have to take care of the logic when fetching from multiple / all brokers

```go
  // Update internally stored URLs, so that we can talk to all brokers
  adminCl. UpdateAPIUrlsFromKubernetesDNS(ctx)

  singleBrokerStatus, err := adminCl.ForBroker(id).GetDebugBundleStatus()
```

Reply:
As per discussion, I have also added `UpdateAPIUrlsFromKubernetesDNS()` method to the Admin API struct.